### PR TITLE
Improve Klaytn support + revise some other ethereum methods

### PR DESCRIPTION
**Pull request by msizov**

### CODE REVIEW

**Review by nikolay-govorov — APPROVED** [review]

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 1]
```diff
@@ -49,7 +49,7 @@ class DefaultEthereumMethods(
             "eth_newPendingTransactionFilter",
         )
 
-        val traceMethods = listOf(
+        var traceMethods = listOf(
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:52]: Let's return to `val` cause we do not reassign this field

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 2]
```diff
@@ -235,6 +323,15 @@ class DefaultEthereumMethods(
             chainUnsupportedMethods(chain) +
             getDrpcVendorMethods(chain) +
             getChainSpecificMethods(chain)
+
+        // add trace method for arbitrum
+        traceMethods.plus(
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:328]: Let's move it to the `getChainSpecificMethods` method where all specific methods are

File: src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt [hunk 3]
```diff
@@ -409,14 +531,24 @@ class DefaultEthereumMethods(
         return json.toByteArray()
     }
 
-    override fun getGroupMethods(groupName: String): Set<String> =
-        when (groupName) {
+    override fun getGroupMethods(groupName: String): Set<String> {
+        val isArbitrum = when (chain) {
```
Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:535]:
```suggestion
override fun getGroupMethods(groupName: String): Set<String> =
        when (groupName) {
            "filter" -> filterMethods
            "trace" -> getTraceMethods()
            "debug" -> debugMethods
            "default" -> getSupportedMethods()
            else -> emptyList()
        }.toSet()

    private fun getTraceMethods(): List<String> {
        return when (chain) {
            Chain.ARBITRUM__MAINNET -> listOf("arb_methods")
            else -> traceMethods
        }
    }
```

Comment (KirillPamPam) [inline src/main/kotlin/io/emeraldpay/dshackle/upstream/calls/DefaultEthereumMethods.kt:535]:
```kotlin
    private fun getTraceMethods(): List<String> {
        return when (chain) {
            Chain.ARBITRUM__MAINNET, ... other chains -> listOf("arb_methods")
            else -> traceMethods
        }
    }
```

**Review by KirillPamPam — APPROVED** [review]

**Pull request merged**